File: rich_game_mechnism/src/com/tw/core/assistentPower/Tool.java
package com.tw.core.assistentPower;

/**
 * Created by pzzheng on 11/16/16.
 */
public class Tool extends AssistentPower{
    public static Tool BLOCK = new Tool(50);
    public static Tool ROBOT_DULL = new Tool(30);
    public static Tool BOMB = new Tool(50);

    Tool(int value) {
        super(value);
    }
}

File: ruby/rich_game_ruby/src/assistencePower/tool.rb
require_relative '../../src/places/hospital'
class Tool
  attr_reader :value

  def initialize(value)
    @value = value
  end

  BLOCK = Tool.new(50)
  def BLOCK.passOn(place, player)
    place.removeTool
    return place.comeHere(player)
  end

  BOMB = Tool.new(50)
  def BOMB.passOn(place, player)
    place.removeTool
    player.stuckFor(Hospital::HOSPITAL_DAYS)
    return player.game.map.hospital.comeHere(player)
  end

  ROBOT_DULL = Tool.new(30)

  def passOn(place, player)
  end
end